test/Types/Interval.purs

```diff
@@ -0,0 +1,184 @@
+module Test.Types.Interval
+  ( suite
+  , eraSummariesFixture
+  , systemStartFixture
+  ) where
+
+import Prelude
+
+import Aeson (decodeJsonString)
+import Control.Monad.Except (throwError)
+import Data.BigInt (fromString) as BigInt
+import Data.Const (Const)
+import Data.Either (Either(Left, Right), either)
+import Data.Maybe (fromJust)
+import Data.Traversable (traverse_)
+import Effect (Effect)
+import Effect.Aff (Aff, error)
+import Mote (MoteT, group, test)
+import Partial.Unsafe (unsafePartial, unsafeCrashWith)
+import QueryM.Ogmios (EraSummaries, SystemStart)
+import Serialization.Address (Slot(Slot))
+import Test.Spec.Assertions (shouldEqual)
+import Types.BigNum (fromInt) as BigNum
+import Types.Interval
+  ( PosixTimeToSlotError(PosixTimeBeforeSystemStart)
+  , POSIXTime(POSIXTime)
+  , posixTimeToSlot
+  , slotToPosixTime
+  )
+
+type TestPlanM a = MoteT (Const Void)
+  (EraSummaries -> SystemStart -> Effect Unit)
+  Aff
+  a
+
+suite :: TestPlanM Unit
+suite = do
+  group "Interval type" do
+    test "Inverse posixTimeToSlot >>> slotToPosixTime " $ testPosixTimeToSlot
+    test "Inverse slotToPosixTime >>> posixTimeToSlot " $ testSlotToPosixTime
+    test "PosixTimeToSlot errors" $ testPosixTimeToSlotError
+
+-- launchAff_ $ runContract testnetConfig $ getEraSummaries >>= encodeAeson >>> stringifyAeson >>> logShow
```

Comment:
There's a few commented-out lines like this

Reply:
I left the comments as guides of how to generate new fixtures, when EraSummaries update/to indicate where it came from. I'll make the comment clearer

Reply:
Again recommending that we have a small script somewhere that does this